pkg/skaffold/deploy/docker/deploy.go

```diff
@@ -138,23 +151,68 @@ func (d *Deployer) deploy(ctx context.Context, out io.Writer, b graph.Artifact)
 		return err
 	}
 
-	containerName := d.getContainerName(ctx, b.ImageName)
+	containerCfg, err := d.containerConfigFromImage(ctx, b.Tag)
+	if err != nil {
+		return err
+	}
+	containerCfg.ExposedPorts = ports
+
+	initContainers, err := d.debugger.TransformImage(ctx, b, containerCfg)
+	if err != nil {
+		return errors.Wrap(err, "transforming image for debugging")
+	}
+
+	for _, c := range initContainers {
+		if d.debugger.HasMount(c.Image) {
+			// skip duplication of init containers
+			continue
+		}
+		if err := d.client.Pull(ctx, out, c.Image); err != nil {
+			return errors.Wrap(err, "pulling init container image")
+		}
+		id, err := d.client.Run(ctx, out, c, dockerutil.ContainerCreateOpts{})
+		if err != nil {
+			return errors.Wrap(err, "creating container in local docker")
+		}
+		r, err := d.client.ContainerInspect(ctx, id)
+		if err != nil {
+			return errors.Wrap(err, "inspecting init container")
+		}
+		if len(r.Mounts) != 1 {
+			olog.Entry(ctx).Warnf("unable to retrieve mount from debug init container: debugging may not work correctly!")
+		}
+		d.debugger.AddSupportMount(c.Image, r.Mounts[0].Name)
+	}
 
+	containerName := d.getContainerName(ctx, b.ImageName)
+	var mounts []mount.Mount
+	for _, m := range d.debugger.SupportMounts() {
+		mounts = append(mounts, m)
+	}
 	opts := dockerutil.ContainerCreateOpts{
 		Name:     containerName,
-		Image:    b.Tag,
 		Network:  d.network,
-		Ports:    ports,
 		Bindings: bindings,
+		Mounts:   mounts,
 	}
-	id, err := d.client.Run(ctx, out, opts)
+
+	id, err := d.client.Run(ctx, out, containerCfg, opts)
 	if err != nil {
 		return errors.Wrap(err, "creating container in local docker")
 	}
 	d.TrackContainerFromBuild(b, tracker.Container{Name: containerName, ID: id})
 	return nil
 }
 
+func (d *Deployer) containerConfigFromImage(ctx context.Context, image string) (*container.Config, error) {
```

Comment:
Should image -> tag?

Reply:
technically it is, I don't really like referring to tagged images as `Tag` though, since providing just the tag without the FQIN would fail (I realize we do this in the `Artifact` but it's not my favorite practice). I'll change to `taggedImage` to be clear